homeassistant/components/airzone_cloud/sensor.py

```diff
@@ -0,0 +1,97 @@
+"""Support for the Airzone Cloud sensors."""
+from __future__ import annotations
+
+from typing import Any, Final
+
+from aioairzone_cloud.const import AZD_HUMIDITY, AZD_NAME, AZD_TEMP, AZD_ZONES
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, UnitOfTemperature
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import AirzoneUpdateCoordinator
+from .entity import AirzoneEntity, AirzoneZoneEntity
+
+ZONE_SENSOR_TYPES: Final[tuple[SensorEntityDescription, ...]] = (
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.TEMPERATURE,
+        key=AZD_TEMP,
+        name="Temperature",
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+    SensorEntityDescription(
+        device_class=SensorDeviceClass.HUMIDITY,
+        key=AZD_HUMIDITY,
+        name="Humidity",
+        native_unit_of_measurement=PERCENTAGE,
+        state_class=SensorStateClass.MEASUREMENT,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Add Airzone Cloud sensors from a config_entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    sensors = []
+    for zone_id, zone_data in coordinator.data[AZD_ZONES].items():
+        for description in ZONE_SENSOR_TYPES:
+            if description.key in zone_data:
+                sensors.append(
+                    AirzoneZoneSensor(
+                        coordinator,
+                        description,
+                        entry,
+                        zone_id,
+                        zone_data,
+                    )
+                )
+
+    async_add_entities(sensors)
+
+
+class AirzoneSensor(AirzoneEntity, SensorEntity):
+    """Define an Airzone Cloud sensor."""
```

Comment:
Is this base class really needed?

Reply:
Yes, I'm gonna create webserver and system entities as in the local integration:
https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/entity.py#L45
https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/entity.py#L78

And I intend to use it as a base class for the different platforms:
https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/binary_sensor.py#L117

Reply:
> Is this base class really needed?

Yes, I'm gonna create webserver and system entities as in the local integration:
https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/sensor.py#L102

https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/sensor.py#L117

https://github.com/home-assistant/core/blob/657d285e8fe346908a1eb618dd4d987bb3190236/homeassistant/components/airzone/sensor.py#L134